synapse/util/patch_inline_callbacks.py

```diff
@@ -37,12 +37,14 @@ def do_patch() -> None:
     if _already_patched:
         return
 
-    def new_inline_callbacks(f):
+    def new_inline_callbacks(f: Callable[..., Generator]) -> Callable[..., Any]:
```

Comment:
In theory this ought to have the same type signature as the original `defer.inlineCallbacks`.
Perhaps we can borrow that?

https://github.com/twisted/twisted/blob/trunk/src/twisted/internet/defer.py#L1827-L1831 :
```py
def inlineCallbacks(f: Callable[..., Generator[Deferred[object], object, T]]) -> Callable[..., Deferred[T]]:
```

which would imply that:
  * `wrapped` below returns a `Deferred[T]`
  * `orig` is a `Callable[..., Deferred[T]]`
  * `_check_yield_points` accepts a `Callable[..., Generator[Deferred[object], object, T]]`
  * and I think `check_yield_points_inner` might return a `Generator[Deferred[object], object, T]`

Reply:
Thanks for this. (I struggled to grok this and clearly just phoned it in with `Any`).

Logic seems sound and mypy seems happy with those annotations!